modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/IngestCommonPlugin.java

```diff
@@ -86,7 +86,8 @@ public Map<String, Processor.Factory> getProcessors(Processor.Parameters paramet
             entry(NetworkDirectionProcessor.TYPE, new NetworkDirectionProcessor.Factory(parameters.scriptService)),
             entry(CommunityIdProcessor.TYPE, new CommunityIdProcessor.Factory()),
             entry(FingerprintProcessor.TYPE, new FingerprintProcessor.Factory()),
-            entry(RegisteredDomainProcessor.TYPE, new RegisteredDomainProcessor.Factory())
+            entry(RegisteredDomainProcessor.TYPE, new RegisteredDomainProcessor.Factory()),
+            entry(RedactProcessor.TYPE, new RedactProcessor.Factory(createGrokThreadWatchdog(parameters)))
```

Comment:
I'm not sure whether we should be invoking `createGrokThreadWatchdog` twice, or merely creating a single value once and passing it to both the `grok` and `redact` processor factories. I'll figure out the answer and let you know.

Reply:
Good catch! The `MatchWatchdog` is used by all the Grok processors on this node and all the `Grok`s that those processors create, it makes sense to use the same watchdog for the `Grok`s created by the redact processor. 

I pushed  https://github.com/elastic/elasticsearch/pull/92951/commits/940c3c1cdcd94473cb09f3a2e4e8efcd77229528